pkg/controllers/argocd_resource_hook_consumer.go

```diff
@@ -0,0 +1,66 @@
+package controllers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/porter-dev/porter-agent/api/server/types"
+	"github.com/porter-dev/porter-agent/internal/models"
+	"github.com/porter-dev/porter-agent/internal/repository"
+)
+
+// ArgoCDResourceHookConsumer contains all information required to consumer an ArgoCDResourceHook
+type ArgoCDResourceHookConsumer struct {
+	Repository *repository.Repository
+}
+
+// NewArgoCDResourceHookConsumer creates an ArgoCDResourceHookConsumer
+func NewArgoCDResourceHookConsumer(repo *repository.Repository) ArgoCDResourceHookConsumer {
+	return ArgoCDResourceHookConsumer{
+		Repository: repo,
+	}
+}
+
+// Consume contains all business logic for consuming an argo resource hook
+func (co ArgoCDResourceHookConsumer) Consume(ctx context.Context, argoEvent types.ArgoCDResourceHook) error {
+	fmt.Printf("Received argo hook: %#v\n", argoEvent)
+
+	ty, ok := resourceHookToEvent[argoEvent.Status]
+	if !ok {
+		return fmt.Errorf("unsupported type %s", argoEvent.Status)
+	}
+
+	ti := time.Now().UTC()
+
+	argoBytes, err := json.Marshal(argoEvent)
+	if err != nil {
+		return fmt.Errorf("error marshalling argo event: %w", err)
+	}
+
+	porterEvent := models.Event{
+		Type:             ty,
+		ReleaseName:      argoEvent.Application,
+		ReleaseNamespace: argoEvent.ApplicationNamespace,
+		Version:          "v1.0.0",
+		Timestamp:        &ti,
+		Data:             argoBytes,
+		UniqueID:         uuid.NewString(),
```

Comment:
Also don't want to mix formats here - the other events just generate a random 16-char via `models.GenerateRandomBytes`: https://github.com/porter-dev/porter-agent/blob/main/internal/models/random.go.